vortex-array/src/compute/min_max.rs

```diff
@@ -0,0 +1,141 @@
+use vortex_dtype::Nullability::NonNullable;
+use vortex_error::{vortex_err, VortexError, VortexResult};
+use vortex_scalar::Scalar;
+
+use crate::stats::{Precision, Stat, Statistics};
+use crate::{Array, Encoding, IntoCanonical};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MinMaxResult {
+    pub min: Scalar,
+    pub max: Scalar,
+}
+
+/// Computes the min and max of an array, returning the (min, max) values
+/// If the array is empty or has only nulls, the result is `None`.
+pub trait MinMaxFn<A> {
+    fn min_max(&self, array: &A) -> VortexResult<Option<MinMaxResult>>;
+}
+
+impl<E: Encoding> MinMaxFn<Array> for E
+where
+    E: MinMaxFn<E::Array>,
+    for<'a> &'a E::Array: TryFrom<&'a Array, Error = VortexError>,
+{
+    fn min_max(&self, array: &Array) -> VortexResult<Option<MinMaxResult>> {
+        let (array_ref, encoding) = array.try_downcast_ref::<E>()?;
+        MinMaxFn::min_max(encoding, array_ref)
+    }
+}
+
+/// Computes the min & max of an array, returning the (min, max) values
+/// The return values are (min, max) scalars, where None indicates that the value is non-existent
+/// (e.g. for an empty array)
+/// The return value dtype is the non-nullable version of the array dtype
```

Comment:
I think the return type should stay the same, but you can guarantee it's not actuall null

Reply:
I think that is more strange behaviour, we know that its non nullable so it should be reflected in the dtype